openstack/dis/v2/streams/requests.go

```diff
@@ -0,0 +1,232 @@
+	StatusTerminated  = "TERMINATED"
+	StatusFrozen      = "FROZEN"
+)
+
+type CreateOpts struct {
+	// Each DIS stream has a unique name. A stream name is 1 to 64 characters in length. Only letters, digits,
+	// hyphens (-), and underscores (_) are allowed.
+	StreamName string `json:"stream_name" required:"true"`
+	// Quantity of the partitions into which data records in the newly created DIS stream will be distributed.
+	// Partitions are the base throughput unit of a DIS stream.
+	// The value range varies depending on the value of stream_type.
+	// If stream_type is not specified or set to COMMON, the value of partition_count is an int from 1 to 50.
+	// If the tenant has created N common partitions, the maximum value of partition_count is 50-N.
+	// If stream_type is set to ADVANCED, the value of partition_count is an int from 1 to 10.
+	// If the tenant has created N advanced partitions, the maximum value of partition_count is 10-N.
+	PartitionCount int `json:"partition_count" required:"true"`
+	// Stream type. Possible values:
+	// COMMON: a common stream. The bandwidth is 1 MB/s.
+	// ADVANCED: an advanced stream. The bandwidth is 5 MB/s.
+	// Default value: COMMON
+	StreamType string `json:"stream_type,omitempty"`
+	// Source data type.
+	// Possible values:
+	// BLOB: a collection of binary data stored as a single entity in a database management system.
+	// JSON: an open-standard file format that uses human-readable text to transmit data objects consisting of attribute–value pairs and array data types.
+	// CSV: a simple text format for storing tabular data in a plain text file.
+	// Default value: BLOB
+	DataType string `json:"data_type,omitempty"`
+	// Period of time for which data is retained in the DIS stream.
+	// Value range: 24 to 72
+	// Unit: hour
+	// Default value: 24
+	// If this parameter is left unspecified, the default value will be used.
+	DataDuration int `json:"data_duration,omitempty"`
+	// Whether to enable automatic Scaling
+	// true: Turn on automatic scale out.
+	// false: Turn off automatic scale in.
+	// Disabled by default.
+	// Default value: false
+	AutoScaleEnabled bool `json:"auto_scale_enabled,omitempty"`
+	// When auto scaling is enabled, the minimum number of slices for auto scaling.
+	// Minimum value: 1
+	AutoScaleMinPartitionCount *int `json:"auto_scale_min_partition_count,omitempty"`
+	// When auto scaling is enabled, the maximum number of slices for auto scaling.
+	AutoScaleMaxPartitionCount *int `json:"auto_scale_max_partition_count,omitempty"`
+	// Source data structure that defines JOSN and CSV formats. It is described in the syntax of Avro.
+	// For details about Avro, see http://avro.apache.org/docs/current/#schemas.
+	// NOTE:
+	// This parameter is mandatory when Dump File Format is Parquet or CarbonData.
+	DataSchema string `json:"data_schema,omitempty"`
+	// Related attributes of CSV format data, such as delimiter
+	CsvProperties CsvProperty `json:"csv_properties,omitempty"`
```

Comment:
The type should be `*CsvProperty`, and for the object name, shall we just use `CsvProperty`?

Reply:
keep it as it is